Game4.py

```diff
@@ -0,0 +1,165 @@
+import os, sys
+import pygame
+from random import randint
+
+from misc import *
+
+
+class Game4Main:
+    """The Main Game4 Class - This class handles the main
+    initialization and creating of the Game."""
+
+    def __init__(self, width=480,height=360):
+        # initialize pygame
+        pygame.init()
+        # set window size
+        self.width = width
+        self.height = height
+        # create screen
+        self.screen = pygame.display.set_mode((self.width, self.height))
+        # create clock
+        self.clock = pygame.time.Clock()
+        self.score = 0
+
+
+    def mainLoop(self):
+        """Main screen for game."""
+        # initialize player
+        count = 0
+        play_len = 25
+        play_x = 100
+        play_y = self.height - play_len
+        player = Player(play_x, play_y, play_len, self.screen)
+
+        # initialize stamina bars
+        P1_stamina_bar = StaminaBar(self.screen,25,"WHITE")
+        P2_stamina_bar = StaminaBar(self.screen,350,"RED")
+
+        # initialize obstacle length, x and y coordinates
+        obs_len = 25
+        obs_x = self.width
+        obs_y = self.height - obs_len
+        obstical_height = self.height - obs_len
+
+        # create list of obstacles
+        obstacles = []
+        new_obstical = False
+        # uncomment and place line below in [] of obstacles = [] to generate random obstacles:
+        # Obstacle(obs_x, obs_y, obs_len, self.screen,'BLUE')
+
+
+        # initialize time variables
+        prev_time = 0
+        obs_dt = 500
+
+        # main event loop
+        while 1:
+            count+=1
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    sys.exit()
+                # check keyboard for obstacle player
+                if event.type == pygame.KEYDOWN:
+                    # z to create obstacle at ground level
+                    if event.key == pygame.K_z:
+                        new_obstical = True
+                        obstical_height = obs_y
+                    # x to create obstacle at second level
+                    if event.key == pygame.K_x:
+                        new_obstical = True
+                        obstical_height = obs_y-25
+                    # c to create obstacle at third level
+                    if event.key == pygame.K_c:
+                        new_obstical = True
+                        obstical_height = obs_y-50
+
+            # check keyboard for main player
+            pressed = pygame.key.get_pressed()
+            # up to jump
+            if pressed[pygame.K_UP] and P1_stamina_bar.bars >= player.jumpcost:
+                if  player.play_y == (360 - player.play_len):
```

Comment:
360 is a “magic number”. Replace it by a constant, or an attribute that is initialized next to wherever this value is coming from.